crates/driver/src/domain/competition/mod.rs

```diff
@@ -52,11 +53,18 @@ pub struct Competition {
     pub mempools: Mempools,
     /// Cached solutions with the most recent solutions at the front.
     pub settlements: Mutex<VecDeque<Settlement>>,
+    pub bad_tokens: Arc<bad_tokens::Detector>,
 }
 
 impl Competition {
     /// Solve an auction as part of this competition.
-    pub async fn solve(&self, auction: &Auction) -> Result<Option<Solved>, Error> {
+    pub async fn solve(&self, auction: Auction) -> Result<Option<Solved>, Error> {
+        let auction = self
```

Comment:
Why not use reference here (`&self...`) instead of line 66?

Reply:
I also thought of bringing this up. Looks like the whole bad token detection logic should not be responsibility of this `/solve` function, but rather part of some Auction preparation logic outside of this function (especially since we already have some logic for altering the original auction we received by autopilot (`pre_processor.prioritize()`)

Reply:
> Why not use reference here (&self...) instead of line 66?

Not sure what you mean. Do you mean why not use `&Auction`? The reason is to avoid cloning all these orders.
Another alternative would be to put the filtering logic onto the `Auction` with `&mut self` but it seems weird that an auction would filter itself based on some component that gets passed in.

> Looks like the whole bad token detection logic should not be responsibility of this /solve function, but rather part of some Auction preparation logic outside of this function (especially since we already have some logic for altering the original auction we received by autopilot (pre_processor.prioritize())

I initially wanted to put it there but the function you referenced is run once per auction for all connected solvers. That means it can only do things that all solvers need. Bad token detection is unique to each solver so it has to happen somewhere else.
I put it into `solve()` because this is logic that is specific to the solver but I agree that it doesn't feel great there either. Seemed like the lesser of 2 evils, though.

Not sure if I addressed your comments completely.

Reply:
I think I didn't described what I meant precisely enough, I was proposing to remove line 66 which is:
```
let auction = &auction;
```
and instead of that use reference operator at line 62:
```
let auction = &self
            .bad_tokens
            .filter_unsupported_orders_in_auction(auction)
            .await;
```

Reply:
Hmm, I assumed that `rustc` would complain if I store a reference to a temporary value like this but apparently that compiles. 👍